packages/mwp-logger-plugin/src/index.js

```diff
@@ -1,19 +1,56 @@
 import logger from './logger';
 
-const onRequestError = (request, err) => {
-	logger.error(
-		{
-			err,
+export function logResponse(request) {
+	const { response, id, server: { app: { logger } } } = request;
+
+	if (response.isBoom) {
+		// response is an Error object
+		logger.error({
+			err: response,
 			...request.raw,
-		},
-		`500 Internal server error: ${err.message}`
-	);
+		});
+	}
+
+	const log = ((response.statusCode >= 500 && logger.error) ||
+		(response.statusCode >= 400 && logger.warn) ||
+		logger.info)
+		.bind(logger);
+
+	const routePluginSettings =
+		request.route.settings.plugins['mwp-logger-plugin'];
+	if (!routePluginSettings || routePluginSettings.enabled) {
+		log({ httpRequest: request, id });
+	}
+
+	return;
+}
+
+const onRequestError = (request, err) => {
```

Comment:
this _might_ result in some duplicate logs, but it was something we had before so I don't want to remove it yet